cli/lib/png2src.js

```diff
@@ -120,37 +120,38 @@ function run(sourceFile) {
         inputColorType: 1,
     });
 
-    const palette = {};
+    const palette = new Map();
     let colorCount = 0;
-    for (let y = 0; y < png.height; ++y) {
-        for (let x = 0; x < png.width; ++x) {
-            const idx = 4 * (png.width * y + x);
-            const r = png.data[idx];
-            const g = png.data[idx + 1];
-            const b = png.data[idx + 2];
-            const a = png.data[idx + 3];
-            const packed = (r << 24) | (g << 16) | (b << 8) | a;
-            let s = palette[packed]
-            if (!s) {
-                if (colorCount >= 4) {
-                    let RGBs = [];
-                    for (let key in palette) {
-                        RGBs.push(`- (R: ${palette[key].r}, G: ${palette[key].g}, B: ${palette[key].b}, A: ${palette[key].a}); first seen at (${palette[key].x}, ${palette[key].y})`)
+    if (png.palette) {
+        colorCount = png.palette.length;
+        if (colorCount >= 4) {
+            throw new Error('Indexed PNG has too many colors: maximum is 4.');
+        }
```

Comment:
This refuses 4 color palette indexed images.

Reply:
Good catch